src/Core/Parser/Patterns.php

```diff
@@ -241,7 +241,7 @@
 					(?:"(?:\\\"|[^"])*")
 					|(?:\'(?:\\\\\'|[^\'])*\')
 				)
-				|(?P<VariableIdentifier>[a-zA-Z0-9\-_.]*[a-zA-Z]+)  # variable identifiers must contain letters (otherwise they are hardcoded numbers)
+				|(?P<VariableIdentifier>[a-zA-Z0-9\-_.]*[a-zA-Z]+[a-zA-Z0-9.]*) # variable identifiers must contain letters (otherwise they are hardcoded numbers)
```

Comment:
I can do the escaping of course, but I concentrated on the bugfix and not adjusting already existing code.

Dot at the end is not really valid, but I currently see no good way to prevent it in this stage of parsing, let me think this through again though. Underscore is debatable, it wasn't before, so I left it out as well.